init-tools.sh

```diff
@@ -98,15 +117,21 @@ if [ ! -e $__DOTNET_PATH ]; then
 
     echo "Installing dotnet cli..."
     __DOTNET_LOCATION="https://dotnetcli.azureedge.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.tar.gz"
-    # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
-    echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> $__init_tools_log
-    if command -v curl > /dev/null; then
-        curl --retry 10 -sSL --create-dirs -o $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-    else
-        wget -q -O $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
-    fi
-    cd $__DOTNET_PATH
-    tar -xf $__DOTNET_PATH/dotnet.tar
+    retries=5
```

Comment:
I think it is probably a good idea to default these in the execute function instead of having them be spread everwhere and then make them be default params to the function.

Reply:
Agreed, I was originally holding off on suggesting `execute` changes so it stays the same as the script in BuildTools, but we should treat this as the master copy.

Reply:
I agree.  I'll move these to the function.